doc/modules/clustering.rst

```diff
@@ -278,6 +285,67 @@ small, as shown in the example and cited reference.
    D. Sculley, *Proceedings of the 19th international conference on World
    wide web* (2010)
 
+.. _k_medoids:
+
+K-Medoids
+=========
+
+
+:class:`KMedoids` is related to the :class:`KMeans` algorithm. While
+:class:`KMeans` tries to minimize the within cluster sum-of-squares,
+:class:`KMedoids` tries to minimize the sum-of-distances between points labeled
+as the medoids of the cluster an all points assigned to that cluster.
+Furthermore, the distance metric used by :class:`KMedoids` can be freely chosen
+while :class:`KMeans` uses a fixed Euclidean distance.
+
+:class:`KMedoids` can be more robust to noise and outliers than :class:`KMeans`
+as e.g. it will choose one of the cluster members as the medoid while
+:class:`KMeans` will move the center of the cluster towards the outlier which
+might in turn move other points away from the cluster centre.
+
+Please note that K-Medoids is different from the K-Medians algorithm as it
+always chooses a point from the dataset as the cluster centre. K-Medians might
+choose a point which is not in the original dataset. For example, given the
+vectors (0,1), (1,0) and (2,2), the Manhattan-distance median is (1,1),
+which does not exist in the original data, and thus cannot be a medoid.
+
+.. topic:: Examples:
+
+ * :ref:`example_cluster_plot_kmedoids_digits.py`: Applying K-Medoids on digits
+ with various distance metrics.
+
+**Algorithm description:**
+:class:`KMedoids` fitting algorithms have different variants, one of them being
+Partitioning Around Medoids (PAM). Currently Scikit-Learn only supports PAM.
+The PAM algorithm uses a greedy search, which may fail to find the global
+optimum. It consists of two alternating steps commonly called the Assignment
+and Update steps (BUILD and SWAP in Kaufmann, Rousseeuw).
+
+PAM works as follows:
+* Initialize: Select ``n_clusters`` from the dataset as the medoids using either
+  a heuristic or random approach (configurable using the ``init`` parameter).
+* Assignment step: assign each element from the dataset to the closest medoid.
+* Update step: For each medoid :math:`m_i` and each data point :math:`x_j`
+  associated to :math:`m_i` swap :math:`m_i` and :math:`o_i` and compute the
+  total cost. Select the medoid :math:`o_i` with the lowest cost.
+* Repease the assignment and update step while the medoids keep changing or
+  maximum number of iterations ``max_iter`` is reached
+
+Total cost is defined as the sum-of-distances from the cluster medoids to the
+medoid.
+
+The complexity of K-Medoids is :math:`O(N^2 k T)` where :math:`N` is the number
+of samples, :math:`T` is the number of iterations and :math:`k` is the number of
+clusters. This makes it more suitable for smaller datasets in comparison to
```

Comment:
Hmm, why? The table says KMedoids -> large clusters, KMeans -> very large :) You'd change  large to ... medium / small?